### PR TITLE
RFC: Bee-rfcs README

### PR DESCRIPTION
**Pull request by jakubcech**
[Rendered](https://github.com/iotaledger/bee-rfcs/blob/7e7c5ef23a57f6f02382becbae52a7f63337ba17/text/0000-bee-rfcs-readme/0000-bee-rfcs-readme.md) (is this how it's done? 😃) 

An RFC proposing we outline a consistent way of writing RFC motivations.

### CODE REVIEW

File: text/0000-bee-rfcs-readme/0000-bee-rfcs-readme.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+# Rationale and alternatives
+
+- The job story approach forces thinkink from the perspective of the person who will be using the product.
+- Standard requirement format was considered, but it allows for too much vagueness.
+- Acceptance criteria format was considered, but that is mainly suitable for products with UI or very UX heavy products. 
+- What is the impact of not doing this?: Not doing this I'm afraid will lead to incosistent motivation specifications and disallow us from quickly validating individual RFCs from the motivation perspective.
+
+# Unresolved questions
+
+- Do we want to enforce this in a format like this? Or should we use a more loose format?
```
Comment (wusyong) [inline text/0000-bee-rfcs-readme/0000-bee-rfcs-readme.md:60]: I think just make this loose format. Let draftsman has some clues on how to write it at least.

Comment (jakubcech) [reply text/0000-bee-rfcs-readme/0000-bee-rfcs-readme.md:60]: I'm all for suggesting and not enforcing at all costs.

File: text/0000-bee-rfcs-readme/0000-bee-rfcs-readme.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+```
+Why are we doing this? What use cases does it support? What is the expected outcome?
+```
+
+With the following:
+```
+Why are we doing this? What use cases does it support? What is the expected outcome?
+
+1. Write a summary of the motivation.
+2. List all the specific use cases that your proposal is trying to address. Write them from the perspective of the person who will be using the software. We recommend using the "Job story" format, which provides context, motivation, and the desired outcome.
```
Comment (SuperFluffy) [inline text/0000-bee-rfcs-readme/0000-bee-rfcs-readme.md]: Replace this with:

2. List all the specific use cases that your proposal is trying to address.
3. Where applicable, write from the perspective of the person who will be using the software, for example using the "Job story" format:

File: text/0000-bee-rfcs-readme/0000-bee-rfcs-readme.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+Example 2:
+When I configure a node, I want to be able to control how much transaction history the node stores, so I can make sure I only store the data I need without incurring additional operational costs.
+
+```
+
+The suggested format is an evolution on a user story format. It may not apply to all RFCs, but I'd encourage everyone to try and think of any functionality with this framing.
+
+# Drawbacks
+
+- The motivation part of RFCs will take longer to write up.
```
Comment (SuperFluffy) [inline text/0000-bee-rfcs-readme/0000-bee-rfcs-readme.md:46]: I don't think that's the case. The suggested changes might actually help to make the process clearer for the writer.

File: text/0000-bee-rfcs-readme/0000-bee-rfcs-readme.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+- Larger RFCs may tackle large amounts of 'jobs'. In which case it might be beneficial to either break down the RFC, or list only higher level motivations.
```
Comment (SuperFluffy) [inline text/0000-bee-rfcs-readme/0000-bee-rfcs-readme.md:47]: This also seems orthogonal to the proposal. If RFCs touch very many different concerns, they should be broken up, if possible.

File: text/0000-bee-rfcs-readme/0000-bee-rfcs-readme.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+- For the current implementation of 'known components', this may seem redundant. If the RFCs motivation is really clear, however, it should not be an issue to write such a motivation.
```
Comment (SuperFluffy) [inline text/0000-bee-rfcs-readme/0000-bee-rfcs-readme.md:48]: Motivating a change is always recommended because what seems obvious to a long-time user of IOTA might not be clear to a newbie. If the particular proposed change doesn't fit the motivation steps as laid out here, people are always free to not adhere to them.

File: text/0000-bee-rfcs-readme/0000-bee-rfcs-readme.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+- If it's an internal component, it may be difficult to think from the user perspective. However, even internal components need to tie into a specific scenario that they allow.
```
Comment (SuperFluffy) [inline text/0000-bee-rfcs-readme/0000-bee-rfcs-readme.md:49]: Hopefully there is always an advantage to the end user, for example in terms of performance, flexibility, or stability, so that part can be motivated.

I agree that it might be difficult to come up with a “user story”. So we shouldn't enforce this too much.

File: text/0000-bee-rfcs-readme/0000-bee-rfcs-readme.md (hunk 1 above)
Comment (SuperFluffy) [reply text/0000-bee-rfcs-readme/0000-bee-rfcs-readme.md:60]: I have changed the wording of the “job story” format somewhat. This steers proposal writers into the right direction, but leaves it flexible enough that people will not be discouraged.

Since we want to evaluate every RFC anyways, we can always point out that a “user story” motivation would have been possible (if that's the case), and request the RFC to be amended thus.

File: text/0000-bee-rfcs-readme/0000-bee-rfcs-readme.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
++ Feature name: `BEE RFCs readme motivation change`
++ Start date: 2019-09-26
++ RFC PR: [iotaledger/bee-rfcs#0000](https://github.com/iotaledger/bee-rfcs/pull/0000)
++ Bee issue: [iotaledger/bee#0000](https://github.com/iotaledger/bee/issues/0000)
+
+# Summary
+
+This RFC proposes a change to the RFC process itself. Particularly the `Motivation` section of the RFC template to put more emphasis on the need to write clear requirements and how to write them.
```
Comment (SuperFluffy) [inline text/0000-bee-rfcs-readme/0000-bee-rfcs-readme.md]: This does not seem to change the RFC process itself. This RFC seems more like a clarification and suggestion as to how to write RFCs.

**Review by SuperFluffy — CHANGES_REQUESTED** [review]
I like the proposed changes.

Other than written in the RFC, I note that this RFC does not seem to change the RFC process itself. The proposed changes to the template seem to rather clarify the intent we are looking for in a new RFC.

As such, I think we could safely close this RFC and merge the changes without the RFC process.


@jakubcech I would ask you to edit `0000-template.md` with your proposed changes (and possibly my suggestions) and commit them to this PR.

**Comment by jakubcech** [comment]
Committed a change to the template itself.
Made changes to the Motivation steps as proposed.

File: text/0000-bee-rfcs-readme/0000-bee-rfcs-readme.md (hunk 2 above)
Comment (jakubcech) [reply text/0000-bee-rfcs-readme/0000-bee-rfcs-readme.md]: Done

**Review by SuperFluffy — APPROVED** [review]

**Comment by SuperFluffy** [comment]
I approve of this proposal, but I don't think it warrants its own RFC because I don't see a fundamental change in the actual RFC procedure.

I personally vote for closing this RFC, submitting the modified template as a new PR and putting the argument into the PR text.

Alternatively, just modify this PR and remove `text/0000-bee-rfcs-readme/0000-bee-rfcs-readme.md`, adding its text to the top comment.

**Comment by jakubcech** [comment]
Created a new PR: 
https://github.com/iotaledger/bee-rfcs/pull/17
